manifests/vhost.pp

```diff
@@ -1853,7 +1853,7 @@
   Optional[Variant[Array[String],String]] $redirectmatch_status                       = undef,
   Optional[Variant[Array[String],String]] $redirectmatch_regexp                       = undef,
   Optional[Variant[Array[String],String]] $redirectmatch_dest                         = undef,
-  Optional[String] $headers                                                           = undef,
+  Optional[Array[String]] $headers                                                    = undef,
```

Comment:
If you're changing it to an array, I'd say drop the `Optional` part and make it an empty array by default. I'd also narrow it to non-empty strings:
```suggestion
  Array[String[1]] $headers                                                    = [],
```

Then you can simplify `templates/vhost/_header.erb` a lot too.

You can also drop the `if $headers` and stick to `! empty($headers)` on line 2263.

Reply:
@ekohl  I thought about that, but realised the patch the way I did for the following reasons:
- I only have the time to actually fix the bug which breaks our puppet runs, I did not want to rework more than really necessary
- every other array type parameter in the module, including $request_headers is an Optional[Array[String]]. I wanted to stay consistent

For those reasons I would like this fix to be accepted. If the datatype should be changed to Array with default [], I would like this seen as an independent rework issue